components/clp-py-utils/clp_py_utils/s3_utils.py

```diff
@@ -104,36 +104,46 @@ def generate_container_auth_options(
     :return: Tuple of (whether aws config mount is needed, credential env_vars to set).
     :raises: ValueError if environment variables are not set correctly.
     """
-    storages_by_component_type: List[Union[S3Storage, FsStorage]] = []
+    output_storages_by_component_type: List[Union[S3Storage, FsStorage]] = []
+    input_storage_needed = False
+
     if component_type in (
         COMPRESSION_SCHEDULER_COMPONENT_NAME,
         COMPRESSION_WORKER_COMPONENT_NAME,
     ):
-        storages_by_component_type = [clp_config.logs_input, clp_config.archive_output.storage]
+        output_storages_by_component_type = [clp_config.archive_output.storage]
+        input_storage_needed = True
     elif component_type in (LOG_VIEWER_WEBUI_COMPONENT_NAME,):
-        storages_by_component_type = [clp_config.stream_output.storage]
+        output_storages_by_component_type = [clp_config.stream_output.storage]
     elif component_type in (
         QUERY_SCHEDULER_COMPONENT_NAME,
         QUERY_WORKER_COMPONENT_NAME,
     ):
-        storages_by_component_type = [
-            clp_config.logs_input,
+        output_storages_by_component_type = [
             clp_config.archive_output.storage,
             clp_config.stream_output.storage,
         ]
+        input_storage_needed = True
```

Comment:
Why's this true for the query scheduler & worker?

Reply:
Thanks for catching this. looks like a mistake made in previous PR.